packages/vee-validate/src/useForm.ts

```diff
@@ -529,9 +530,10 @@ export function useForm<
 
             if (result.valid && typeof fn === 'function') {
               const controlled = deepCopy(controlledValues.value);
-              let submittedValues = (onlyControlled ? controlled : values) as unknown as TOutput;
+              const submittedValues = (onlyControlled ? controlled : values) as unknown as TOutput;
+
               if (result.values) {
-                submittedValues = result.values;
+                Object.assign(submittedValues, result.values);
               }
```

Comment:
This might break some people using `yup.strip` and other kind of object transformations as you are merging the result of the current form values with the parsed ones. Instead of just returning the parsed ones.

I will double check against that tonight. But is it necessary as far as I can tell you are building the value object with a loop anyways:

```ts
if (validation.value) {
  setInPath(values, validation.key, validation.value);
}
```

 If not then maybe we can force the `validate` function on the field-level to always return a value to avoid doing a merge.

Reply:
I tried a few different variants of this. I get several errors in tests if I force the field-level validate to always return a value and then do the simple assignment here.

```diff
 FAIL  packages/vee-validate/tests/Form.spec.ts > <Form /> > unmounted fields gets unregistered and their submitted values are kept if configured on the field level
AssertionError: expected last "spy" call to have been called with [ { drink: [ 'Tea', 'Coke' ], …(2) } ]

- Expected
+ Received

  Array [
    Object {
      "drink": Array [
        "Tea",
        "Coke",
      ],
-     "nested": Object {
-       "field": "12",
-     },
-     "non-nested.field": "12",
    },
  ]

```